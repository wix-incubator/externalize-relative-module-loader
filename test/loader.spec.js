const path = require('path');
const webpack = require('./helpers/compiler');

describe('Loader', () => {
  const config = {
    loader: {
      test: /[\\/]node_modules[\\/]lodash/
    },
    externals: ['lodash']
  };

  it('should transform imports for slashed imports', async () => {
    const stats = await webpack('with-slash.js', config);
    const {source} = stats.toJson().modules[1];

    expect(source).toMatchSnapshot();
  });

  it('should transform imports for slashed imports', async () => {
    const stats = await webpack('with-dot.js', config);
    const {source} = stats.toJson().modules[1];

    expect(source).toMatchSnapshot();
  });
});
