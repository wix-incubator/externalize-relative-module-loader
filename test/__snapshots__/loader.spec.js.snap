// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should transform imports for slashed imports 1`] = `"module.exports = require('lodash').map"`;

exports[`Loader should transform imports for slashed imports 2`] = `"module.exports = require('lodash').camelCase"`;
